packages/office-ui-fabric-react/src/components/Dialog/Dialog.tsx

```diff
@@ -79,21 +76,24 @@ export class Dialog extends BaseComponent<IDialogProps, IDialogState> {
         isClickableOutsideFocusTrap={ isClickableOutsideFocusTrap }
         isDarkOverlay={ isDarkOverlay }
         isOpen={ isOpen }
-        onDismiss={ onDismiss }
         onDismissed={ onDismissed }
         onLayerDidMount={ onLayerDidMount }
         responsiveMode={ responsiveMode }
+        { ...modalProps }
+        onDismiss={ onDismiss } // pull from contentProps.onDismiss once onDismiss is deprecated
```

Comment:
I wasn't quite sure how to tackle this one. I know that DialogContent.Props needs to accept onDismiss. Do we support both, and pass down either/or?